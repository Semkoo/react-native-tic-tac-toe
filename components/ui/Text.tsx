import { Text as RNText, StyleSheet } from 'react-native';
import React from 'react';

export default function Text({ children, style, ...props }: React.ComponentProps<typeof RNText>) {
  return (
    <RNText style={[styles.text, style]} {...props}>
      {children}
    </RNText>
  );
}

const styles = StyleSheet.create({
  text: {
    fontFamily: 'SpaceMono',
  },
});
