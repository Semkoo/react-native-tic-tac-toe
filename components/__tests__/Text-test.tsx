import * as React from 'react';
import renderer from 'react-test-renderer';

import Text from '../ui/Text';

it('renders correctly', () => {
  const tree = renderer.create(<Text>Test</Text>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with children', () => {
  const tree = renderer.create(<Text>Test</Text>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with children and style', () => {
  const tree = renderer.create(<Text style={{ color: 'red' }}>Test</Text>).toJSON();
  expect(tree).toMatchSnapshot();
});
