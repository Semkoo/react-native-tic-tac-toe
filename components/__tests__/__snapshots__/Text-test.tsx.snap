// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "SpaceMono",
      },
      undefined,
    ]
  }
>
  Test
</Text>
`;

exports[`renders correctly with children 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "SpaceMono",
      },
      undefined,
    ]
  }
>
  Test
</Text>
`;

exports[`renders correctly with children and style 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "SpaceMono",
      },
      {
        "color": "red",
      },
    ]
  }
>
  Test
</Text>
`;
